File: backend/app/models/analytics.py
from typing import Literal

from pydantic import BaseModel, Field


class AnalyticsRange(BaseModel):
    start: str
    end: str
    timezone: str


class AnalyticsFocusByDay(BaseModel):
    date: str
    focusMinutes: int


class AnalyticsFocusByTask(BaseModel):
    taskLabel: str
    focusMinutes: int


class AnalyticsDistractingApp(BaseModel):
    applicationName: str
    estimatedMinutes: float


class AnalyticsSnapshot(BaseModel):
    range: AnalyticsRange
    focusTodayMinutes: int
    totalFocusMinutes: int
    completedSessions: int
    interruptionCount: int
    averageSessionMinutes: float | None
    breakCompliancePercent: float | None
    focusByDay: list[AnalyticsFocusByDay]
    focusByTask: list[AnalyticsFocusByTask]
    topDistractingApps: list[AnalyticsDistractingApp]

    def to_evidence_payload(self) -> dict:
        return self.model_dump()

File: src/renderer/types/analytics.ts
export interface AnalyticsFocusByDay {
  date: string;
  focusMinutes: number;
}

export interface AnalyticsFocusByTask {
  taskLabel: string;
  focusMinutes: number;
}

export interface AnalyticsDistractingApp {
  applicationName: string;
  estimatedMinutes: number;
}

export interface AnalyticsSnapshot {
  range: {
    start: string;
    end: string;
    timezone: string;
  };
  focusTodayMinutes: number;
  totalFocusMinutes: number;
  completedSessions: number;
  interruptionCount: number;
  averageSessionMinutes: number | null;
  breakCompliancePercent: number | null;
  focusByDay: AnalyticsFocusByDay[];
  focusByTask: AnalyticsFocusByTask[];
  topDistractingApps: AnalyticsDistractingApp[];
}
